Mathlib/RingTheory/Coalgebra/TensorProduct.lean

```diff
@@ -0,0 +1,192 @@
+/-
+Copyright (c) 2024 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.Algebra.Category.CoalgebraCat.ComonEquivalence
+
+/-!
+# Tensor products of coalgebras
+
+Given two `R`-coalgebras `M, N`, we can define a natural comultiplication map
+`Δ : M ⊗[R] N → (M ⊗[R] N) ⊗[R] (M ⊗[R] N)` and counit map `ε : M ⊗[R] N → R` induced by
+the comultiplication and counit maps of `M` and `N`. These `Δ, ε` are declared as linear maps
+in `TensorProduct.instCoalgebraStruct` in `Mathlib.RingTheory.Coalgebra.Basic`.
+
+In this file we show that `Δ, ε` satisfy the axioms of a coalgebra, and also define other data
+in the monoidal structure on `R`-coalgebras, like the tensor product of two coalgebra morphisms
+as a coalgebra morphism.
+
+## Implementation notes
+
+We keep the linear maps underlying `Δ, ε` and other definitions in this file syntactically equal
+to the corresponding definitions for tensor products of modules in the hope that this makes
+life easier. However, to fill in prop fields, we use the API in
+`Mathlib.Algebra.Category.CoalgebraCat.ComonEquivalence`. That file defines the monoidal category
+structure on coalgebras induced by an equivalence with comonoid objects in the category of modules,
+`CoalgebraCat.instMonoidalCategoryAux`, but we do not declare this as an instance - we just use it
+to prove things. Then, we use the definitions in this file to make a monoidal category instance on
+`CoalgebraCat R` in `Mathlib.Algebra.Category.CoalgebraCat.Monoidal` that has simpler data.
+
+-/
```

Comment:
Please add a note about the universe monomorphism resulting from using the category theory API, possibly including an invitation for someone in the future to generalize this!

Reply:
> possibly including an invitation for someone in the future to generalize this!

The more general version exists already at https://github.com/eric-wieser/FLT/blob/8764e28f03cd0d03f673e1288f47fc7a665207de/FLT/for_mathlib/Coalgebra/TensorProduct.lean

Reply:
Thanks, done